torchao/csrc/cuda/tensor_core_tiled_layout/libtorch.cpp

```diff
@@ -0,0 +1,245 @@
+//     if (arg.isInt()) {
+//       ministack[idx] = reinterpret_cast<void *>(arg.toInt());
+//     } else if (arg.isTensor()) {
+//       const at::Tensor& tensor = arg.toTensor();
+//       AtenTensorHandle ath = torch::aot_inductor::tensor_pointer_to_tensor_handle(&tensor);
+//       ministack[idx] = reinterpret_cast<void *>(ath);
+//     } else {
+//       TORCH_CHECK(false, "Other types of IValues not yet handled!");
+//     }
+//   }
+
+//   // second function is going to take a stack of void*, cast them to our
+//   // schema values for now, and run the function and modify the void* stack
+//   voidyvoid_boxed_ATH_unpack_tensor_core_tiled_layout(ministack, num_arguments,
+//                                                     num_returns);
+
+//   // now pop all inputs on stack. if we pop earlier, Tensors would go out of scope
+//   // before calling the function
+//   torch::jit::drop(stack, num_arguments);
+
+//   // read the output from the end of the stack and wrap that back into
+//   // IValue from void*?
+//   for (auto idx = 0; idx < num_returns; idx++) {
+//     const c10::TypePtr& ret_type = schema.returns()[idx].type();
+//     if (*ret_type == *c10::getTypePtr<at::Tensor>()) {
+//       AtenTensorHandle ret_ath = reinterpret_cast<AtenTensorHandle>( ministack[num_arguments + idx]);
+//       at::Tensor out = *torch::aot_inductor::tensor_handle_to_tensor_pointer(ret_ath);
+//       torch::jit::push(stack, c10::IValue(out));
+//     } else {
+//       TORCH_CHECK(false, "Only Tensor return types are currently supported!");
+//     }
+//   }
+
+//   free(ministack);
+// }
+
+// TORCH_LIBRARY_IMPL(torchao, CUDA, m) {
+//   // m.impl("torchao::unpack_tensor_core_tiled_layout",
+//   //        &_unpack_tensor_core_tiled_layout);
+//   m.impl("torchao::unpack_tensor_core_tiled_layout",
+//          torch::CppFunction::makeFromBoxedFunction<
+//              boxed_unpack_tensor_core_tiled_layout>());
+//   // m.impl("torchao::dequantize_tensor_core_tiled_layout",
+//   // &_dequantize_tensor_core_tiled_layout);
+//   m.impl("torchao::dequantize_tensor_core_tiled_layout",
+//          torch::CppFunction::makeFromBoxedFunction<
+//              boxed_dequantize_tensor_core_tiled_layout>());
+// }
+
+class StableLibrary::TorchLibraryOpaque {
+public:
+    TorchLibraryOpaque(StableLibrary::Kind kind, std::string ns, std::optional<c10::DispatchKey> k, const char* file, uint32_t line)
+        : library_(kind, ns, k, file, line) {}
+private:
+    torch::Library library_; // Actual Library object
+};
+
+StableLibrary::StableLibrary(StableLibrary::Kind kind, std::string ns, std::optional<c10::DispatchKey> k, const char* file, uint32_t line)
+    : lib_(&TorchLibraryOpaque(Library::Kind::IMPL, ns, k, file, line)) {}
+
+StableLibrary& StableLibrary::impl(std::string name, void (*fn)(void **, int64_t, int64_t)) {
+  auto boxed_function = [fn](const c10::OperatorHandle &op, torch::jit::Stack *stack) {
```

Comment:
are these, `torch::jit:*` and IValue, considered stable?

Reply:
no! I haven't gotten a chance to document what I'm trying to do, but everything in libtorch.cpp should eventually make its way back to libtorch (not stable!!)

everything in libtorch.h is supposed to be stable.